thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/api/controller/QuestionController.java

```diff
@@ -21,12 +23,12 @@ public class QuestionController {
     private final PictureQuestionService pictureQuestionService;
 
     @GetMapping("/definition/random")
-    public DefinitionQuestionResponse getRandomDefinitionQuestion(@PathVariable int categoryId) {
-        return definitionQuestionService.generateQuestion(categoryId, null);
+    public DefinitionQuestionResponse getRandomDefinitionQuestion(@Current UserAuthDetails userAuthDetails, @PathVariable int categoryId) {
+        return definitionQuestionService.generateQuestion(categoryId, userAuthDetails);
     }
 
     @GetMapping("/picture/random")
-    public PictureQuestionResponse getRandomPictureQuestion(@PathVariable int categoryId) {
-        return pictureQuestionService.generateQuestion(categoryId, null);
+    public PictureQuestionResponse getRandomPictureQuestion(@Current UserAuthDetails userAuthDetails, @PathVariable int categoryId) {
+        return pictureQuestionService.generateQuestion(categoryId, userAuthDetails);
```

Comment:
I left it as `random`, because it is still random :) . I limited the pool of terms to these the closest to user's rank, but I still get a random term from them.